### PR TITLE
Feat #437 Parameters in the Reference Data Library should modifyable

### LINKED ISSUE #437: [Reference Data] Parameters in the Reference Data Library should modifyable
Eventually, it should be possible to modify, create and delete parameter types from the RDL

### PR DESCRIPTION
**Pull request by joao4all**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Closes #437 Parameters in the Reference Data Library should modifyable

### CODE REVIEW

File: COMETwebapp/Components/ReferenceData/ParameterTypes/ComponentsTable.razor.cs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        private void OnTextChanged(string text)
+        {
+            this.Dimension = text;
+            var dimensions = text.Split(",").Select(int.Parse).ToList();
+
+            var arrayParameterType = (ArrayParameterType)this.ParameterType;
+
+            if (arrayParameterType.Dimension.Count > dimensions.Count)
+            {
+                var j = 0;
```
Comment (antoineatstariongroup) [inline COMETwebapp/Components/ReferenceData/ParameterTypes/ComponentsTable.razor.cs:111]: Can you rename 'j' with a better name, like dimensionIndex ?

File: COMETwebapp/Components/ReferenceData/ParameterTypes/ComponentsTable.razor.cs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                    j++;
+                }
+            }
+
+            if (arrayParameterType.Dimension.Count >= dimensions.Count)
+            {
+                return;
+            }
+
+            var k = 0;
```
Comment (antoineatstariongroup) [inline COMETwebapp/Components/ReferenceData/ParameterTypes/ComponentsTable.razor.cs:129]: same there

File: COMETwebapp/Components/ReferenceData/ParameterTypes/ComponentsTable.razor.cs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            {
+                if (k >= arrayParameterType.Dimension.Count)
+                {
+                    arrayParameterType.Dimension.Add(dimension);
+                }
+
+                k++;
+            }
+
+            //update the contained component according with the updated dimensions
```
Comment (antoineatstariongroup) [inline COMETwebapp/Components/ReferenceData/ParameterTypes/ComponentsTable.razor.cs:141]: I assume that WIP here?

File: COMETwebapp/Validators/ReferenceData/ParameterTypes/GenericValidators/ParameterTypeValidator.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    /// <summary>
+    /// A class to validate the <see cref="ParameterType" />
+    /// </summary>
+    public class ParameterTypeValidator : AbstractValidator<ParameterType>
+    {
+        /// <summary>
+        /// Instantiates a new <see cref="ParameterTypeValidator" />
+        /// </summary>
+        public ParameterTypeValidator(IValidationService validationService)
```
Comment (antoineatstariongroup) [inline COMETwebapp/Validators/ReferenceData/ParameterTypes/GenericValidators/ParameterTypeValidator.cs:42]: For all validator, missing parameter documentation for IValidationService

**Review by antoineatstariongroup — CHANGES_REQUESTED** [review]
Minor documentation improvement

File: COMETwebapp/Components/ReferenceData/ParameterTypes/ComponentsTable.razor.cs (hunk 3 above)
Comment (joao4all) [reply COMETwebapp/Components/ReferenceData/ParameterTypes/ComponentsTable.razor.cs:141]: Yeah, Sam is going to create a new ticket for that. It's about the implementation of ArrayParameterType.

Comment (samatstariongroup) [reply COMETwebapp/Components/ReferenceData/ParameterTypes/ComponentsTable.razor.cs:141]: see https://github.com/STARIONGROUP/COMET-WEB-Community-Edition/issues/602

**Pull request merged**